Classes/Exporter/EventsExporter.php

```diff
@@ -0,0 +1,390 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package jweiland/events2.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace JWeiland\Events2\Exporter;
+
+use GuzzleHttp\Exception\GuzzleException;
+use JWeiland\Events2\Domain\Factory\TimeFactory;
+use JWeiland\Events2\Domain\Model\Event;
+use JWeiland\Events2\Domain\Model\Link;
+use JWeiland\Events2\Domain\Model\Location;
+use JWeiland\Events2\Domain\Model\Time;
+use JWeiland\Events2\Service\EventService;
+use JWeiland\Events2\Utility\DateTimeUtility;
+use Psr\Http\Message\ResponseInterface;
+use TYPO3\CMS\Core\Exception\SiteNotFoundException;
+use TYPO3\CMS\Core\Http\JsonResponse;
+use TYPO3\CMS\Core\Http\RequestFactory;
+use TYPO3\CMS\Core\LinkHandling\Exception\UnknownLinkHandlerException;
+use TYPO3\CMS\Core\LinkHandling\LinkService;
+use TYPO3\CMS\Core\Site\SiteFinder;
+
+class EventsExporter
+{
+    public function __construct(
+        protected readonly EventService $eventService,
+        protected readonly SiteFinder $siteFinder,
+        protected readonly TimeFactory $timeFactory,
+        protected readonly RequestFactory $requestFactory,
+        protected readonly LinkService $linkService,
+        protected readonly DateTimeUtility $dateTimeUtility,
+    ) {}
+
+    public function export(ExporterConfiguration $configuration): ResponseInterface
+    {
+        $preparedEvents = $this->getPreparedEvents(
```

Comment:
@sfroemkenjw The export method has an internal processing step (getPreparedEvents) and an external HTTP request. Would it be nice to separate these responsibilities into two methods: one for preparing data and another for the actual request?
It would make it easier to test each part independently and allow the potential reuse of getPreparedEvents.